lib/App.js

```diff
@@ -235,7 +235,7 @@ App.prototype.component = function(name, constructor, isDependency) {
 
   // Calling app.component() overrides existing views or components. Prevent
   // dependencies from doing this without warning
-  if (isDependency && currentView && !serializedViews) {
+  if (isDependency && currentView && typeof serializedViews !== 'function') {
```

Comment:
Optional - Since `typeof serializedViews` won't change after it's required at module-level, it'd be very slightly faster to cache `typeof serializedViews !== 'function'` once at module level.

It'd also be nice for the future to leave a note as to why the special exception is there.

Reply:
modified to be less hacky by adding property to view explicitly in deserializing